### DISCUSSION TITLE
Error "There was an errror connecting to the server at the specified address." after upgrade (3.12.0 to 4.4.2).

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#6001] [Q&A] [answered]
Hi,

I upgraded Mirth Connect from 3.12.0 to 4.4.2. Now as I start Administrator Launcher (1.4.1) and click the Launch button, apparently I am able to log in ("Authenticated" is briefly visible in the login screen) but then the login screen displays "There was an error connecting to the server at the specified address. Please verify that the server is up and running." in red, and stops there.

I've compared the "3.12.0 mirth.properties" file with the "4.4.2 mirth.properties" file and found that they differ in two places:

**I.**
```
https.client.protocols
https.server.protocols
https.ciphersuites
```

and since providing the old values here might actually solve the problem (as described in https://forums.mirthproject.io/forum/mirth-connect/support/12484-there-was-an-error-connecting-to-the-server) I commented out the new values and replaced them with the old ones:

```
#https.client.protocols = TLSv1.3,TLSv1.2
#https.server.protocols = TLSv1.3,TLSv1.2,SSLv2Hello
#https.ciphersuites = TLS_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_AES_256_GCM_SHA384,TLS_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_DSS_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_DSS_WITH_AES_128_GCM_SHA256,TLS_EMPTY_RENEGOTIATION_INFO_SCSV
#https.ephemeraldhkeysize = 2048
https.client.protocols = TLSv1.3,TLSv1.2,TLSv1.1
https.server.protocols = TLSv1.3,TLSv1.2,TLSv1.1,SSLv2Hello
https.ciphersuites = TLS_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_AES_256_GCM_SHA384,TLS_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDH_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDH_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_DHE_DSS_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDH_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_RSA_WITH_AES_128_GCM_SHA256,TLS_DHE_DSS_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384,TLS_RSA_WITH_AES_256_CBC_SHA256,TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA384,TLS_ECDH_RSA_WITH_AES_256_CBC_SHA384,TLS_DHE_RSA_WITH_AES_256_CBC_SHA256,TLS_DHE_DSS_WITH_AES_256_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA,TLS_ECDH_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDH_RSA_WITH_AES_256_CBC_SHA,TLS_DHE_RSA_WITH_AES_256_CBC_SHA,TLS_DHE_DSS_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,TLS_RSA_WITH_AES_128_CBC_SHA256,TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDH_RSA_WITH_AES_128_CBC_SHA256,TLS_DHE_RSA_WITH_AES_128_CBC_SHA256,TLS_DHE_DSS_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDH_RSA_WITH_AES_128_CBC_SHA,TLS_DHE_RSA_WITH_AES_128_CBC_SHA,TLS_DHE_DSS_WITH_AES_128_CBC_SHA,TLS_EMPTY_RENEGOTIATION_INFO_SCSV
https.ephemeraldhkeysize = 2048

```

But now I got this error even before I could log in...

```
java.lang.OutOfMemoryError: Java heap space
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.concurrent.FutureTask.report(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.get(Unknown Source)
	at com.mirth.connect.client.launcher.f.a(SourceFile:416)
	at com.mirth.connect.client.launcher.f.a(SourceFile:364)
	at com.mirth.connect.client.launcher.MirthClientLauncher.run(SourceFile:1326)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.OutOfMemoryError: Java heap space

```
...which looks even worse to me so I changed things back.

**II.**
These lines were added at the bottom:

```
# The algorithm to use when decrypting old message content.
encryption.fallback.algorithm = AES

# Allows old digest values to be verified
digest.fallback.algorithm = SHA256
digest.fallback.iterations = 1000
digest.fallback.usepbe = 0
```

This is described in https://github.com/nextgenhealthcare/connect/wiki/4.4.0---Upgrade-Guide and I am not quite sure what they do in detail but I don't think they are related to the error.

The machine I installed on runs OpenJDK 18. Recently I was able to install Mirth Connect 4.4.1 on a machine with OpenJDK 18 so I don't think OpenJDK 18 is to blame: 4.4.2 doesn't differ that much from 4.4.1 according to the specs. 
Since the error also mentioned "at the specified address" I checked whether the ports might already be in use with `netstat`, but this looks fine as well:

```
  TCP    0.0.0.0:8080           0.0.0.0:0              LISTENING
  TCP    [::]:8080              [::]:0                 LISTENING
  TCP    0.0.0.0:8443           0.0.0.0:0              LISTENING
  TCP    [::]:8443              [::]:0                 LISTENING
```

I don't know what else to look into to solve the error. Anyone any ideas?

**Comment by jonbartels** [comment]
>  then the login screen displays "There was an error connecting to the server at the specified address. Please verify that the server is up and running." in red, and stops there.

Anything in `mirth.log` when this happens?

Can you hit the web console at `https://yourserver:8443` and log in there? This will tell you if it is a server issue or a client issue.

When you log in from the Launcher, turn on "Show Java Console". Any output there?

BTW - your writeup was excellent. Thank you for capturing the details, I hope the community can help

**Reply by Coooz** [reply]
"BTW - your writeup was excellent."
Thank you! 😄

**Comment by Coooz** [comment, marked as answer]
I just looked at the issue with a colleague who has some more knowlegde on sql server databases. After puzzling for some time, eventually we did a reinstall of Mirth Connect, and provided the simplest connection string possible. The original connection string had `;encrypt=true;trustServerCertificate=true` appended to it, we left that out. And that worked.
Then we tried connecting using the original connection string once more, including the `;encrypt=true;trustServerCertificate=true` part, and then that worked as well.

I'm not quite sure what happened here... 😕

**Comment by rjkroll** [comment]
Sounds like the Mirth service was either stopped or hung and rebooting fixed it.
